projects/packages/publicize/src/class-publicize.php

```diff
@@ -578,9 +585,15 @@ public function flag_post_for_publicize( $new_status, $old_status, $post ) {
 	public function test_connection( $service_name, $connection ) {
 		$id = $this->get_connection_id( $connection );
 
+		if ( array_key_exists( $id, $this->test_connection_results ) ) {
+			return $this->test_connection_results[ $id ];
+		}
+
 		$xml = new Jetpack_IXR_Client();
 		$xml->query( 'jetpack.testPublicizeConnection', $id );
 
+		$this->test_connection_results[ $id ] = ! $xml->isError();
```

Comment:
This will cache `false` if there is an error. Future calls to this method for the same connection will return that `false` instead of the WP_Error returned in the first call. This does not comply with the `@return` tag PHPDoc in the [abstract definition](https://github.com/Automattic/jetpack/blob/5d34e751e6e5ef76d7b576c5449eb5849e9cc305/projects/packages/publicize/src/class-publicize-base.php#L711) of the `test_connection` method.

Reply:
Good catch!
Fixed now.